maps/rivers_and_bridges.go

```diff
@@ -0,0 +1,386 @@
+		isHazard := false
+		for _, h := range lastBoardState.Hazards {
+			if p == h {
+				isHazard = true
+				break
+			}
+		}
+
+		if !isHazard {
+			totallyUnoccupiedPoints = append(totallyUnoccupiedPoints, p)
+		}
+	}
+
+	return totallyUnoccupiedPoints
+}
+
+func setupRiverAndBridgesBoard(startingPositions [][]rules.Point, hazards []rules.Point, lastBoardState *rules.BoardState, settings rules.Settings, editor Editor) error {
+	width := lastBoardState.Width
+	height := lastBoardState.Height
+
+	numSnakes := len(lastBoardState.Snakes)
+	if numSnakes == 0 {
+		return rules.RulesetError("too few snakes - at least one snake must be present")
+	}
+
+	rand := settings.GetRand(0)
+
+	snakeIDs := make([]string, 0, len(lastBoardState.Snakes))
+	for _, snake := range lastBoardState.Snakes {
+		snakeIDs = append(snakeIDs, snake.ID)
+	}
+
+	tempBoardState := rules.NewBoardState(width, height)
+	tempBoardState.Snakes = make([]rules.Snake, len(snakeIDs))
+
+	for i := 0; i < len(snakeIDs); i++ {
+		tempBoardState.Snakes[i] = rules.Snake{
+			ID:     snakeIDs[i],
+			Health: rules.SnakeMaxHealth,
+		}
+	}
+	err := rules.PlaceSnakesInQuadrants(rand, tempBoardState, startingPositions)
+	if err != nil {
+		return err
+	}
+
+	err = rules.PlaceFoodFixed(rand, tempBoardState)
+	if err != nil {
+		return err
+	}
+
+	// Copy food from temp board state
+	for _, f := range tempBoardState.Food {
+		// skip the center food
+		if f.X == lastBoardState.Width/2 && f.Y == lastBoardState.Height/2 {
+			continue
+		}
+		editor.AddFood(f)
+	}
+
+	// Copy snakes from temp board state
+	for _, snake := range tempBoardState.Snakes {
+		editor.PlaceSnake(snake.ID, snake.Body, snake.Health)
+	}
+
+	for _, p := range hazards {
+		editor.AddHazard(p)
+	}
+
+	return nil
+}
+
+func placeRiverAndBridgesFood(lastBoardState *rules.BoardState, settings rules.Settings, editor Editor) error {
+	rand := settings.GetRand(lastBoardState.Turn)
+
+	foodNeeded := checkFoodNeedingPlacement(rand, settings, lastBoardState)
+	if foodNeeded > 0 {
+		pts := getUnoccupiedPoints(lastBoardState)
+		placeFoodRandomlyAtPositions(rand, lastBoardState, editor, foodNeeded, pts)
+	}
+
+	return nil
+}
+
+type RiverAndBridgesMediumHazardsMap struct{}
+
+func (m RiverAndBridgesMediumHazardsMap) ID() string {
+	return "hz_rivers_bridges"
+}
+
+func (m RiverAndBridgesMediumHazardsMap) Meta() Metadata {
+	return Metadata{
+		Name: "hz_rivers_bridges",
+		Description: `Creates fixed maps that have a lake of hazard in the middle with rivers going in the cardinal directions.
+Each river has one or two 1-square "bridges" over them`,
+		Author:     "Battlesnake",
+		Version:    1,
+		MinPlayers: 1,
+		MaxPlayers: 8,
+		BoardSizes: FixedSizes(Dimensions{11, 11}),
+		Tags:       []string{TAG_FOOD_PLACEMENT, TAG_HAZARD_PLACEMENT, TAG_SNAKE_PLACEMENT},
+	}
+}
+
+func (m RiverAndBridgesMediumHazardsMap) SetupBoard(lastBoardState *rules.BoardState, settings rules.Settings, editor Editor) error {
```

Comment:
We might want to have these return an error if the board size isn't the supported one. Otherwise it draws the map regardless of whether it fits on the board.

Reply:
I was wondering about this, i didn't know if the map system had that built in or if we need to do standard checks at the start of SetupBoard() on every map

Reply:
It's not built in currently. Right now it's sort of the maps choice whether to enforce the supported sizes and validate settings.
I think if it's going to produce "wrong" behaviour like off-grid hazards vs. just "funny-looking" games, that might be the reason to enforce it in the map. Unless something's changed, the error message should be passed back to the play UI, so you can at least see why a game wasn't created.